homeassistant/components/proxmoxve/binary_sensor.py

```diff
@@ -1,112 +1,95 @@
-        self._name = None
+                # unfound container case
+                if coordinator_data is None:
+                    continue
 
-        self._state = None
+                container_name = coordinator_data["name"]
+                container_status = create_binary_sensor(
+                    coordinator, host_name, node_name, container_id, container_name
+                )
+                sensors.append(container_status)
 
-    @property
-    def name(self):
-        """Return the name of the entity."""
-        return self._name
+    add_entities(sensors)
 
-    @property
-    def is_on(self):
-        """Return true if VM/container is running."""
-        return self._state
 
-    @property
-    def device_state_attributes(self):
-        """Return device attributes of the entity."""
-        return {
-            "node": self._item_node,
-            "vmid": self._item_id,
-            "vmname": self._vmname,
-            "type": self._item_type.name,
-            ATTR_ATTRIBUTION: ATTRIBUTION,
-        }
-
-    def update(self):
-        """Check if the VM/Container is running."""
-        item = self.poll_item()
-
-        if item is None:
-            _LOGGER.warning("Failed to poll VM/container %s", self._item_id)
-            return
-
-        self._state = item["status"] == "running"
-
-    def poll_item(self):
-        """Find the VM/Container with the set item_id."""
-        items = (
-            self._proxmox_client.get_api_client()
-            .nodes(self._item_node)
-            .get(self._item_type.name)
-        )
-        item = next(
-            (item for item in items if item["vmid"] == str(self._item_id)), None
-        )
+def create_binary_sensor(coordinator, host_name, node_name, vm_id, name):
+    """Create a binary sensor based on the given data."""
+    return ProxmoxBinarySensor(
+        coordinator=coordinator,
+        unique_id=f"proxmox_{node_name}_{vm_id}_running",
+        name=f"{node_name}_{name}_running",
+        icon="",
+        host_name=host_name,
+        node_name=node_name,
+        vm_id=vm_id,
+    )
+
 
-        if item is None:
-            _LOGGER.warning("Couldn't find VM/Container with the ID %s", self._item_id)
-            return None
+class ProxmoxBinarySensor(ProxmoxEntity):
+    """A binary sensor for reading Proxmox VE data."""
 
-        if self._vmname is None:
-            self._vmname = item["name"]
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        unique_id,
+        name,
+        icon,
+        host_name,
+        node_name,
+        vm_id,
+    ):
+        """Create the binary sensor for vms or containers."""
+        super().__init__(
+            coordinator, unique_id, name, icon, host_name, node_name, vm_id
+        )
 
-        if self._name is None:
-            self._name = f"{self._item_node} {self._vmname} running"
+        self._state = None
 
-        return item
+    @property
+    def state(self):
```

Comment:
Binary sensor entities should not implement the `state` property but the `is_on` property. Please change this.